### DISCUSSION TITLE
[LAMMPS] -  Neighbor list overflow Error

### DISCUSSION THREAD

**Discussion by AnuragKr** [deepmodeling/deepmd-kit#2025] [Q&A] [answered]
I trained model using data [link](https://dplibrary.deepmd.net/#/project_details?project_id=202109.002).
After obtaining compressed DP potential I am attempting to do simulation with 10000,100000 atoms but I am getting error for above two dataset size. It is working fine for 1000 atoms.

```lammps
#Starting
units metal
atom_style atomic
boundary p p p
read_data data/ice_pimd_100000.dat
pair_style deepmd ../../ice_pimd_single_gpu_graph-compress.pb
pair_coeff * *

timestep 0.00005

group Oatoms type 1
group Hatoms type 2

#minimize 1.0e-10 1.0e-10 100000 100000

fix 1 all npt temp 300 300 0.0005 iso 1.0 1.0 0.005

thermo_style custom step temp ke pe etotal press vol
thermo 100
thermo_modify format line "%d %.6e %.6e %.6e %.6e %.6e %.6e %.6e %.6e %.6e"

dump my_dump all custom 100 result/ice_pimd_10000_fs.xyz id type x y z vx vy vz
dump_modify my_dump sort id
dump my_dump2 all atom 100 result/ice_pimd_10000.lammps-dump-text

run 10000
``` 

Error
![Screenshot 2022-10-26 at 6 36 12 PM](https://user-images.githubusercontent.com/10074710/198033622-4e980c1b-bfb1-4ae7-94e2-388edccab08c.png)

Please suggest what appropriate neighbor parameter I should use.I tried following combination--

```lammps
neighbor 0.3 bin
neigh_modify every 1 delay 0 check yes one 1000 page 10000
```

```lammps
neighbor 0.3 bin
neigh_modify one 10000000
```

**Comment by njzjz** [comment]
I don't think 2000 neighbors/atom is expected. Please check the data.

**Reply by AnuragKr** [reply]
Like I am creating data through lammps simulation and using it as it is for the next simulation.
Ex-First I ran for 1000 loops with 1000 atoms to create 11000 data and planning to do the same for creating 100000, etc.

Is this the right way to create data?

**Reply by njzjz** [reply]
Did you change the box when you added more atoms?

**Reply by AnuragKr** [reply]
No
Current Box structure --
```lammps
0.0 12.44470 xlo xhi
0.0 12.44470 ylo yhi
0.0 12.44470 zlo zhi
```
Now I tried to change the coordinates by extracting min,max from data itself.
Updated Coordinates

```lammps
-0.869175 13.4509 xlo xhi
-0.869175 13.4509 ylo yhi
-0.869175 13.4509 zlo zhi
```
Vizualization--
![Screenshot 2022-10-28 at 12 51 28 PM](https://user-images.githubusercontent.com/10074710/198528359-090f65f4-3a66-4618-a4c8-be3e67783339.png)

I ran a simulation with an updated simulation box coordinated but still, the error remain the same.

Is it the correct way to define simulation box coordinates or if you can suggest any other method to - for atoms of size - 1000,10000,100000 etc?

**Reply by njzjz** [reply]
Obviously your atoms are too close to each other.

**Reply by AnuragKr** [reply]
Please suggest ways to generate data and define simulation box coordinates.

**Reply by njzjz** [reply]
You should translate atoms when copying atoms (meaning no atoms overlapping).

**Reply by AnuragKr** [reply]
Could you please explain from a code perspective - How I should translate atoms and any specific changes I need to do in lammps code or when I am creating lammps data files ?

**Reply by njzjz** [reply]
You can refer LAMMPS [replicate command](https://docs.lammps.org/replicate.html#replicate-command) or [`ase.Atoms.repeat`](https://wiki.fysik.dtu.dk/ase/ase/atoms.html#ase.Atoms.repeat).

**Reply by AnuragKr** [reply]
Thanks, It solved my issue.

**Comment by AnuragKr** [comment, marked as answer]
Using the replicate command we can create more no. of atoms for simulation. 
Ex - Initially I had 500 atoms 
      For 1000 atoms -  replicate 2 1 1
     For 10000 atoms -  replicate 2 2 5
